lib/query.js

```diff
@@ -80,7 +80,18 @@ Query.prototype.handleRowDescription = function(msg) {
 };
 
 Query.prototype.handleDataRow = function(msg) {
-  var row = this._result.parseRow(msg.fields);
+  var row;
+
+  try {
+    row = this._result.parseRow(msg.fields);
+  } catch (err) {
+    if(this.callback) {
+      return this.callback(err);
```

Comment:
This will call the callback at least twice if there are errors: once for each errored row, and then once to incorrectly indicate a successful completion, with errored rows missing.